test/Grpc.IntegrationTests/LargePayloadTests.cs

```diff
@@ -0,0 +1,647 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using Microsoft.DurableTask.Client;
+using Microsoft.DurableTask;
+using Microsoft.DurableTask.Converters;
+using Microsoft.DurableTask.Worker;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit.Abstractions;
+
+namespace Microsoft.DurableTask.Grpc.Tests;
+
+public class LargePayloadTests(ITestOutputHelper output, GrpcSidecarFixture sidecarFixture) : IntegrationTestBase(output, sidecarFixture)
+{
+    // Validates client externalizes a large orchestration input and worker resolves it.
+    [Fact]
+    public async Task LargeOrchestrationInputAndOutputAndCustomStatus()
+    {
+        string largeInput = new string('A', 1024 * 1024); // 1MB
+        TaskName orchestratorName = nameof(LargeOrchestrationInputAndOutputAndCustomStatus);
+
+        InMemoryPayloadStore fakeStore = new InMemoryPayloadStore();
+
+        await using HostTestLifetime server = await this.StartWorkerAsync(
+            worker =>
+            {
+                worker.AddTasks(tasks => tasks.AddOrchestratorFunc<string, string>(
+                    orchestratorName,
+                    (ctx, input) =>
+                    {
+                        ctx.SetCustomStatus(largeInput);
+                        return Task.FromResult(input + input);
+                    }));
+
+                worker.UseExternalizedPayloads();
+
+                worker.Services.AddSingleton<PayloadStore>(fakeStore);
+            },
+            client =>
+            {
+                client.UseExternalizedPayloads();
+
+                // Override store with in-memory test double
+                client.Services.AddSingleton<PayloadStore>(fakeStore);
+            },
+            services =>
+            {
+                services.AddExternalizedPayloadStore(opts =>
+                {
+                    opts.ExternalizeThresholdBytes = 1024;
+                    opts.ContainerName = "test";
+                    opts.ConnectionString = "UseDevelopmentStorage=true";
```

Comment:
Nit, you use a direct `PayloadStore` registration and the new extension method. Can you just use the extension and make sure it still works?